### DISCUSSION TITLE
NR01=1 with latest gopro 11 labs update worse

### DISCUSSION THREAD

**Discussion by lyricsdude** [gopro/labs#535] [General]
Hi,

I was very happy with the fine grain/noise in the previous gopro labs version from march when NR01 was set to 1. But in the latest version there's no fine grain anymore but clear noise reduction applied which can't be dealt with properly in post production editing. What's the reason for this? I can't install a previous firmware and now I'm stuck and can't continue filming my episodes.

Can you please release a fix for this? Thanks

**Comment by dnewman-gpsw** [comment]
Show me an example as NR01 is working for me and I think everyone else.  Shoot a short clip with it enabled and share that clip here.

**Reply by lyricsdude** [reply]
![march-firmware](https://github.com/gopro/labs/assets/91627562/fdf16e06-2acb-4386-a46f-858625dcdf9e)
this is the march firmware. very fine grain and no sign of any noise reduction

![august](https://github.com/gopro/labs/assets/91627562/f8e0d340-f5ad-4061-894b-4bc4ff0761c0)
and this is the latest firmware. clearly patterns of noise reduction

**Reply by dnewman-gpsw** [reply]
I don't see it.  Both look to have NR off and nothing has changed with this feature.

**Reply by dnewman-gpsw** [reply]
Current version on HERO11  (Pixel level crop form 5.3Kp24)
![image](https://github.com/gopro/labs/assets/10856084/c9232c0f-978a-4554-99da-987272c52d11)

**Reply by lyricsdude** [reply]
yes this is what i mean. in your example again very clear noise reduction patterns with NR01=1. Why can't I install the march gopro labs firmware? I did a factory reset and still I can't install it.

**Reply by dnewman-gpsw** [reply]
Noise reduction is correctly disabled, but camera tuning maybe be different. I tried with the March 2.10.70 release and there are subtle differences (although lighting conditions may have changes too.)

![image](https://github.com/gopro/labs/assets/10856084/85faa502-f1c8-4b40-8a42-55abff97ae9d)

It not possible to roll back to March version in public release camera, I had to use a developer only procedure to do it. Why, I don't know, mainline firmware change, not a Labs change. The code for NR01 is the same, but I will look to see if there are any other noise reductions added.  I personally prefer the current look, but there does seem to be more noise in the March release.

**Reply by dnewman-gpsw** [reply]
I don't see code coding or tuning difference. In the above differences between Labs version was likely just the amount of light in the scene.

**Reply by lyricsdude** [reply]
thanks for doing these tests. exactly what you say there's more noise in the march release. if noise reduction is 1 i prefer all noise reduction turned off. i don't like processing a processed video, the end result will be worse. the noise reduction in davinci resolve works much better with natural grain/noise.

is the result the same for gopro 12 btw?

**Reply by lyricsdude** [reply]
<img width="493" alt="Screenshot 2023-09-16 at 1 39 59 PM" src="https://github.com/gopro/labs/assets/91627562/b3a5cef4-4a49-4b38-9c66-49b65e95b77b">
this is my shirt with noise reduction applied in davinci resolve in post. the weird pattern is unacceptable. in the old firmware my shirt silky smooth when i apply noise reduction in post

**Reply by Deadcode85** [reply]
Could you please share the 2.10.70 official labs FW? I also dont like the new pattern in the noise in the latest FW. With the Hero 10 it was possible to roll back previous FW.
I prefer the noisier but more natural look of the previous FW.
Im glad i find this thread, i triple checked my settings if i turned off the NR properly in the past couple of days.

**Reply by Deadcode85** [reply]
Ok now i have the 2.10.70 labs, but as you said David, the update fails. Is there a way for us to roll back? The only reason i upgraded from Hero 10 is the NR off modul, and i feel like kinda cheated. I had an experience with a Hero 11 with a march FW, and i satisfied with that

**Reply by lyricsdude** [reply]
if i click the factory reset button it just resets my settings, still the latest firmware is installed. how can i get a true factory reset as what it truly means?

**Reply by dnewman-gpsw** [reply]
Sorry you can't install 2.10.70 anymore, same with 2.10 or earlier, there some core change outside of Labs development.

I can see there is something still enabled, yet I don't find any code different on Labs NR control.  However, I can reproduce it, and therefore something to improve for 2.20.70 (in development now.)

**Reply by Deadcode85** [reply]
So can you tell our request the development team? We want noisy image! :D At least we want to have an option to have a completly noisy image.

**Reply by lyricsdude** [reply]
it would be great if we can get the same results as the previous firmware

**Reply by lyricsdude** [reply]
is it confirmed that in the next firmware update we will be able to turn the noise reduction totally off again?

**Reply by dnewman-gpsw** [reply]
Not confirmed, as the cause for the small difference has not been determined.

**Reply by dnewman-gpsw** [reply]
While the Labs code had not changed, I expect some other new denoising engine was enabled in core FW.  So here are some tests build that turn off more than before (only at NR01=1, not any other level.)  I look forward to your feedback.

HERO11 Labs 2.12.71:  https://www.dropbox.com/t/A5mfO3DANNqTBrHn
HERO12 Labs 1.10.71:  https://www.dropbox.com/t/xu9Px5QUPEtKPvQ1

**Reply by lyricsdude** [reply]
thank you so much for your effort. it looks great again! just tested it :)

**Reply by lyricsdude** [reply]
the temporal noise reduction in davinci resolve works much better again with this unprocessed footage :)

**Reply by Deadcode85** [reply]
Thank you very much, i will too test it, but only tomorrow. Is it possible to rollback 2.12.70 this time, or should i do comparison shots before updating?

**Reply by lyricsdude** [reply]
you will see within 1 second it's back to normal again :)

**Reply by dnewman-gpsw** [reply]
You can switch between versions 2.12.70/71 and future I expect.

**Reply by Deadcode85** [reply]
And yes, it's properly noisy again, thank you very much!! And as you told NR01=2 is the "new" noise pattern if someone likes it better. We are happy!

**Reply by kouyoumin** [reply]
Will there be a independent control, say, NR02,  for that new denoise engine?

**Reply by dnewman-gpsw** [reply]
That was the original idea, NR02 would for more subtle denoise controls.  However this was a regression, so fixing it under NR01 made sense.

**Reply by dtrtuser** [reply]
Hi there,

Did this NR01 fix made it to the newly released (November 2023) HERO11 v2.20.70 version? Can't find mention of that in the release notes.

Thanks!

**Reply by dnewman-gpsw** [reply]
It is in there, and I will add to the release notes

**Reply by dtrtuser** [reply]
Awesome, thank you!

**Comment by YuriL1951** [comment]
To turn off the noise reduction, do you need QR NR01=2? Or just NR02?

**Reply by dnewman-gpsw** [reply]
oMNR01=1  <- temporally turns it off (for that power up session)
!MNR01=1  <- permanently turns it off.

=2 is less off
=1-99 is a percentage the noise reduction is active.
=0 normal camera operation
